backend/de.metas.payment.esr/src/main/java/de/metas/payment/spi/impl/AbstractESRPaymentStringParser.java

```diff
@@ -77,17 +78,7 @@ protected final BigDecimal extractAmountFromString(final Properties ctx, final S
 	{
 		final String amountString = Util.replaceNonDigitCharsWithZero(amountStringWithPosibleSpaces); // 04551
 
-		BigDecimal amount = BigDecimal.ZERO;
-		try
-		{
-			amount = new BigDecimal(amountString);
-		}
-		catch (final NumberFormatException e)
-		{
-			final String wrongNumberFormatAmount = Services.get(IMsgBL.class).getMsg(ctx, ERR_WRONG_NUMBER_FORMAT_AMOUNT, new Object[] { amountString });
-			collectedErrors.add(wrongNumberFormatAmount);
-		}
-		return amount;
+		return NumberUtils.createBigDecimal(amountString);
```

Comment:
what happens in case it fails?

don't you have to collect errors to `collectedErrors`?
if not, then why the param is there?

anyways, get rid of `ctx`

Reply:
ahhh, and pls use de.metas.common.util.NumberUtils and not apache commons.